### PR TITLE
Clarify support of UTF-8 and other encodings

### PR DESCRIPTION
**Pull request by Shane32**
See #229

### CODE REVIEW

File: spec/GraphQLOverHTTP.md [hunk 1]
```diff
@@ -193,10 +193,10 @@ And for a _GraphQL response_:
 For details of the shapes of these JSON payloads, please see
 [Request](#sec-Request) and [Response](#sec-Response).
 
-If the media type in a `Content-Type` or `Accept` header includes encoding
-information, then the encoding MUST be `utf-8` (e.g.
-`Content-Type: application/graphql-response+json; charset=utf-8`). If encoding
-information is not included then `utf-8` MUST be assumed.
+If the media type in a `Content-Type` or `Accept` header does not
+include encoding information, then `utf-8` MUST be assumed (e.g.
+`Content-Type: application/graphql-response+json; charset=utf-8`).
```
Comment (benjie) [inline spec/GraphQLOverHTTP.md:198]: This example doesn't seem to fit any more
```suggestion
include encoding information, then `utf-8` MUST be assumed (e.g.
for header `Content-Type: application/graphql-response+json`, UTF-8 encoding would be assumed).
```

**Review by benjie — CHANGES_REQUESTED** [review]
Good work, generally in favour of these changes 👍

File: spec/GraphQLOverHTTP.md [hunk 2]
```diff
@@ -438,6 +440,9 @@ A server SHOULD support requests which accept the
 `application/graphql-response+json` media type (as indicated by the `Accept`
 header).
 
+A server MUST support the `utf-8` encoding for requests which accept a JSON media type,
+such as the `application/json` or `application/graphql-response+json` media types.
```
[the remark this answers is not in the document]
Comment (Shane32) [reply spec/GraphQLOverHTTP.md:444]: FYI, I tried to word the paragraph in the same manner as the previous two paragraphs, which use the phrase "accept", as this does relate to the `Accept` HTTP header.

If I understand your concern, there's "Accept" which is the HTTP header regarding the response by the server, versus "accept" which is the server accepting a request from the client.  Is that correct?  If so, should we change any of the prior paragraphs as well?  Perhaps I should simply add the phrase "(as indicated by the `Accept` header)"?

Here's some ideas for this paragraph:

```suggestion
A server MUST support the `utf-8` encoding for requests which accept a JSON media type,
such as the `application/json` or `application/graphql-response+json` media types (as indicated by
the `Accept` header).
```

```suggestion
A server MUST support the `utf-8` encoding for requests which return a supported JSON media type,
such as the `application/json` or `application/graphql-response+json` media types (as indicated by
the `Accept` header).
```

I'm open to suggestions.

File: spec/GraphQLOverHTTP.md (hunk 1 above)
Comment (Shane32) [reply spec/GraphQLOverHTTP.md:198]: :+1:

File: spec/GraphQLOverHTTP.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (benjie) [reply spec/GraphQLOverHTTP.md:444]: Thinking about it more, I don't think we need nor want this paragraph. If you support `application/graphql-response+json` right now but only when using `ISO-8859-1` then that's okay. After the watershed you must support it with UTF-8 encoding (I believe this is stated elsewhere?) but that's a specific media-type and encoding combo we explicitly dictate. If you support `application/flibble+json; encoding=ISO-8859-1` I don't think we should dictate this must also be UTF-8.

[the remark this answers is not in the document]
Comment (Shane32) [reply spec/GraphQLOverHTTP.md]: Sounds good; take a look now.

[the remark this answers is not in the document]
Comment (Shane32) [reply spec/GraphQLOverHTTP.md]: So I think this puts us at:

POST requests' `Content-Type`:
- `application/json` must be supported
- `application/json` must be assumed when not specified
- `utf-8` must be assumed for `application/json` media type without encoding specified
- No explicit statement mandating support of `application/json` with `utf-8` encoding, but this is implied by the following statement:

> Request encoding with media type application/json is supported by every compliant server.

`Accept` header:
- `application/json` must be supported indefinitely
- `application/json` must be assumed until waterfall
- `application/graphql-response+json` must be supported after waterfall
- `application/graphql-response+json` must be assumed after waterfall
- `utf-8` is assumed for requests that accept `application/json` or `application/graphql-response+json` without encoding specified
- `application/graphql-response+json; charset=utf-8` must be supported after waterfall

**Comment by Shane32** [comment]
@benjie Let me know if there's anything you're waiting for me on here.  (I believe I made changes pursuant to your last comments...)

**Review by benjie — CHANGES_REQUESTED** [review]
Looking good, I've made some suggestions to tighten it up a little. (Sorry for the delay, I'm 200 GitHub notifications behind where I should be 😅 )

File: spec/GraphQLOverHTTP.md [hunk 3]
```diff
@@ -249,7 +250,7 @@ client gets something useful, it should indicate the media types it supports.
 If the client supplies an `Accept` header, the client SHOULD include the media
 type `application/graphql-response+json` in the `Accept` header.
 
-Note: From 1st Jan 2025, every _server_ and _client_ must support
+Note: From 1st Jan 2025, every _server_ and _client_ MUST support
```
[the remark this answers is not in the document]
Comment (Shane32) [reply spec/GraphQLOverHTTP.md:253]:
```suggestion
Note: From 1st Jan 2025, every _server_ and _client_ must support
```

File: INTERESTED_DEVELOPERS.md [hunk 4]
```diff
@@ -85,4 +88,4 @@ yourself by username.
 
 ### CC Helper
 
-`@abernix @balshor @benjie @deinok @erikwittern @jaydenseric @michaelstaib @mike-marcacci @mmatsa @sjparsons @spawnia @sungam3r @enisdenjo`
+`@abernix @balshor @benjie @deinok @erikwittern @jaydenseric @michaelstaib @mike-marcacci @mmatsa @shane32 @sjparsons @spawnia @sungam3r @enisdenjo`
```
[the remark this answers is not in the document]
Comment (Shane32) [reply INTERESTED_DEVELOPERS.md:91]: - https://github.com/graphql/graphql-over-http/pull/236

File: INTERESTED_DEVELOPERS.md [hunk 5]
```diff
@@ -66,6 +66,9 @@ yourself by username.
 - @mmatsa
   - Company/Project/Repo: https://github.com/graphql/libgraphqlparser
   - Reason: Interested in client/server in C++
+- @shane32
+  - Company/Project/Repo: https://github.com/graphql-dotnet/server
+  - Reason: Interested in a common HTTP spec and common subscriptions spec
```
Comment (Shane32) [inline INTERESTED_DEVELOPERS.md:71]:
```suggestion
```

File: INTERESTED_DEVELOPERS.md (hunk 4 above)
Comment (Shane32) [inline INTERESTED_DEVELOPERS.md:91]:
```suggestion
`@abernix @balshor @benjie @deinok @erikwittern @jaydenseric @michaelstaib @mike-marcacci @mmatsa @sjparsons @spawnia @sungam3r @enisdenjo`
```

**Comment by Shane32** [comment]
@benjie Thanks!! Your suggestions made the spec easier to read than what I had.

**Review by benjie — APPROVED** [review]
This is a normative change - we've changed from requiring that `application/json` and `application/graphql-request+json` use the UTF8 encoding to requiring that the UTF8 encoding must be supported and is assumed in the absence of explicit encoding, but that other encodings are allowed (but not required).

I approve this change; however it will require consensus to get merged.

**Comment by benjie** [comment]
@abernix @balshor @benjie @deinok @erikwittern @jaydenseric @michaelstaib @mike-marcacci @mmatsa @shane32 @sjparsons @spawnia @sungam3r @touchstone117 @enisdenjo

Hey folks; normative change. Please either approve, or state your hesitations. Thanks! (If you think we should discuss this at a GraphQL-over-HTTP WG, that can be arranged.)

**Comment by benjie** [comment]
@Shane32 Please format the spec with prettier 🙏

**Review by spawnia — APPROVED** [review]

**Review by Touchstone117 — APPROVED** [review]

**Review by michaelstaib — APPROVED** [review]

**Pull request merged**